examples/pytorch_imagenet_resnet50.py

```diff
@@ -156,13 +161,24 @@ def train(epoch):
             if args.cuda:
                 data, target = data.cuda(), target.cuda()
             optimizer.zero_grad()
-            output = model(data)
-            loss = F.cross_entropy(output, target)
-            loss.backward()
+            # Split batch into sub-batches and sum gradients for all samples
+            sub_batch_size = int(data.size(0) // args.backward_steps)
+            batch_loss = 0
+            batch_output = []
+            for i in range(0, data.size(0), sub_batch_size):
+                data_step = data[i:i + sub_batch_size]
+                target_step = target[i:i + sub_batch_size]
+                output = model(data_step)
+                batch_output.append(output)
+                loss = F.cross_entropy(output, target_step)
+                batch_loss += loss / args.accum_grad
+                loss.backward()
+            # Gradient is averaged by args.backward_steps by the optimizer
             optimizer.step()
 
-            train_loss.update(loss)
-            train_accuracy.update(accuracy(output, target))
+            batch_output = torch.cat(batch_output, dim=0)
+            train_loss.update(batch_loss.item())
+            train_accuracy.update(accuracy(batch_output, target))
```

Comment:
Can push up accuracy computation into the loop as well, then no need to collect outputs and concat the tensor.